test/framework/toy_build.py

```diff
@@ -271,6 +278,88 @@ def test_toy_broken(self):
         # cleanup
         shutil.rmtree(tmpdir)
 
+    def detect_log_file(self, tmp_logpath):
+        log_files = list(pathlib.Path(tmp_logpath).glob("**/*.log"))
+
+        self.assertTrue(len(log_files) >= 1, 'Log files generated')
+        log_file = log_files[0]
+        self.assertTrue(len(log_files) == 1, f"Log file '{log_file}' is unique")
+
+        return log_file
+
+    def assert_build_files_copied(self, buildpath, errorlogpath):
+        buildir = pathlib.Path(buildpath)
+        errorlogdir = pathlib.Path(errorlogpath)
+        iso_date_pattern = r'????????-??????'
+        for file in buildir.iterdir():
+            file_relative_path = file.relative_to(buildir)
+            file_copies = list(errorlogdir.glob(f"toy-0.0/{iso_date_pattern}/{file_relative_path}"))
+            self.assertTrue(len(file_copies) == 1, f"Unique copy of toy build file '{file}' made")
+            file_copy = file_copies[0]
+
+            if file_copy.is_file():
+                msg = f"File '{file}' copied succesfully"
+                self.assertTrue(filecmp.cmp(str(file), str(file_copy), shallow=False), msg)
+
+    def assert_log_files_copied(self, log_file, errorlogpath):
+        file_name = log_file.name
+        saved_log_files = list(pathlib.Path(errorlogpath).glob(f"**/{file_name}"))
+        self.assertTrue(len(saved_log_files) == 1, f"Unique copy of log file '{log_file}' made")
+        for saved_log_file in saved_log_files:
+            msg = f"Log file '{log_file}' copied succesfully"
+            self.assertTrue(filecmp.cmp(str(log_file), str(saved_log_file), shallow=False), msg)
+
+    def assert_error_reported(self, outtxt, output_regexs):
+        for regex_pattern in output_regexs:
+            regex = re.compile(regex_pattern, re.M)
+            msg_stdout = "Pattern %s found in full test report: %s" % (regex.pattern, outtxt)
+            self.assertTrue(regex.search(outtxt), msg_stdout)
+
+    def check_errorlog(self, output_regexs, outtxt, tmp_logpath, buildpath, errorlogpath, errorartifactpath):
+        log_file = self.detect_log_file(tmp_logpath)
+
+        self.assert_log_files_copied(log_file, errorlogpath)
+        self.assert_build_files_copied(buildpath, errorartifactpath)
+        self.assert_error_reported(outtxt, output_regexs)
+
+        with open(f"{log_file}", 'r') as p_log_file:
+            self.assert_error_reported(p_log_file.read(), output_regexs)
+
+    def test_toy_broken_compilation(self):
+        """Test whether log files and the build directory are copied to a permanent location after a failed
+           compilation."""
+        with tempfile.TemporaryDirectory() as tmpdir, \
+            tempfile.TemporaryDirectory() as tmp_logdir, \
+            tempfile.TemporaryDirectory() as tmp_artifactdir, \
+            tempfile.TemporaryDirectory() as tmp_easyconfig_dir:
```

Comment:
I think this works:
```suggestion
                tempfile.TemporaryDirectory() as tmp_logdir, \
                tempfile.TemporaryDirectory() as tmp_artifactdir, \
                tempfile.TemporaryDirectory() as tmp_easyconfig_dir:
```

Or maybe use a single parent folder instead. `self.test_prefix` is a directory created for each test case and removed afterwards

Reply:
Generated a single temporary directory, and created all other directories in the temporary directory.